test/unit/vfocus/iterations.spec.js

```diff
@@ -0,0 +1,150 @@
+var path = require('path');
+var _ = require('lodash');
+var chai = require('chai');
+var expect = chai.expect;
+
+var h = require('virtual-hyperscript');
+var VText = require('vtree/vtext');
+var VFocus = require(path.resolve(process.cwd(), 'src/vfocus'));
+
+describe('VFocus - Iterations', function() {
+
+
+  describe('forEach', function() {
+    var forEachTree, forEachTreeFocus;
+    beforeEach(function() {
+      forEachTree = h(
+        'div#0', [
+          h('p#1', [
+            h('b#2'),
+            h('i#3')
+          ]),
+          h('p#4')
+      ]);
+    
+      forEachTreeFocus = new VFocus(forEachTree);
+    });
+
+
+    it('iterates through the right nodes in the right order', function(){
+      var nodesIds = [];
+      var expectedNodeIds = ['0', '1', '2', '3', '4'];
+      forEachTreeFocus.forEach(node => nodesIds.push(node.vNode.properties.id));
+
+      expect(nodesIds).to.deep.equal(expectedNodeIds);
+    });
+
+  });
+
+
+  describe('flatten', function() {
+    var flattenTree, rootVFocus, firstVFocus, secondVFocus, thirdVFocus, fourthVFocus;
+    beforeEach(function() {
+      flattenTree = h(
+        'div#0', [
+          h('p#1', [
+            h('b#2'),
+            h('i#3')
+          ]),
+          h('p#4')
+      ]);
+
+      rootVFocus = new VFocus(flattenTree);
+      firstVFocus = rootVFocus.next();
+      secondVFocus = firstVFocus.next();
+      thirdVFocus = secondVFocus.next();
+      fourthVFocus = thirdVFocus.next();
+    });
+
+
+    it('returns the right list of nodes in the right order', function(){
+      var returnedNodeList = rootVFocus.flatten();
+      var expectedNodeList = [rootVFocus, firstVFocus, secondVFocus, thirdVFocus, fourthVFocus];
+
+      expect(returnedNodeList).to.deep.equal(expectedNodeList);
+    });
+
+  });
+
+
+  describe('takeWhile', function() {
+    var takeWhileTree, rootVFocus, firstVFocus, secondVFocus, thirdVFocus;
+    beforeEach(function() {
+      takeWhileTree = h(
+        'div#0', [
+          h('p#1', [
+            h('b#2'),
+            h('i#3')
+          ]),
+          h('p')
+      ]);
+
+      rootVFocus = new VFocus(takeWhileTree);
+      firstVFocus = rootVFocus.next();
+      secondVFocus = firstVFocus.next();
+      thirdVFocus = secondVFocus.next();
+    });
+
+    it('returns the right list of node that have and id', function(){
```

Comment:
Maybe change this sentence to something more general like: 'takes nodes until the predicate is false"